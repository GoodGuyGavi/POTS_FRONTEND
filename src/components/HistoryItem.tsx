import React, { Fragment } from "react";
import ButtonApp from "./Button";
import { ListGroupItem, Row, Col } from "react-bootstrap";
import { List } from "semantic-ui-react";

export interface IHistoryItemProps {
  list?: any;
  buttonTitle?: any;
}

const HistoryItem = (props: IHistoryItemProps) => {
  const { buttonTitle } = props;
  return (
    <Fragment>
      {props.list.map((h: any) => (
        <List>
          <Row>
            <Col md={2}>{h.hisItem}</Col>
            <Col md={2}>
              <ButtonApp buttonTitle={buttonTitle} variant="secondary" />
            </Col>
          </Row>
        </List>
      ))}
    </Fragment>
  );
};

export default HistoryItem;
