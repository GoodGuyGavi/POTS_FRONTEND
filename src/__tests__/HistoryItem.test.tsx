/**
 * @format
 */

import React from "react";
import { shallow } from "enzyme";
import HistoryItem from "../components/HistoryItem";

const hisList = [
  {
    key: 1,
    hisItem: "Some History"
  },
  {
    key: 2,
    hisItem: "Some History"
  }
];

describe("History Item", () => {
  test("renders correctly", () => {
    expect(
      shallow(<HistoryItem list={hisList} buttonTitle="View" />)
    ).toMatchSnapshot();
  });
});
