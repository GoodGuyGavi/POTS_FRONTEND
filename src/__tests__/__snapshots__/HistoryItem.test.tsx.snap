// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History Item renders correctly 1`] = `ShallowWrapper {}`;
